source/adapters/level_zero/memory.cpp

```diff
@@ -1390,6 +1410,13 @@ ur_result_t ur_queue_handle_legacy_t_::enqueueUSMMemcpy2D(
   // (versus compute engine).
   bool PreferCopyEngine = !IsDevicePointer(Queue->Context, Src) ||
                           !IsDevicePointer(Queue->Context, Dst);
+  // For better performance, Copy Engines are not preferred given Shared
+  // pointers on DG2.
+  if ((IsSharedPointer(Queue->Context, Src) ||
+       IsSharedPointer(Queue->Context, Dst)) &&
+      Queue->Device->isDG2()) {
```

Comment:
The order of these conditions should be reversed. `Queue->Device->isDG2()` is just reading a variable (quick), whereas `IsSharedPointer` needs to call into the driver (how expensive is this operation). I'd be worried about impacting performance on non-DG2 systems.

Reply:
it is a very small operation, but I see your point with regards to the check for DG2 first, then the pointer type. We already incur that cost in the previous line for isDevicePointer, but we can reorder to get some time back on other platforms.

Reply:
Hm, ok, if you think it's OK the way it is, we can leave it as is for now (to save time on re-spinning CI), and revisit it later.

Reply:
Updated in the recent commit.